xarray/coding/times.py

```diff
@@ -203,18 +238,50 @@ def _infer_time_units_from_diff(unique_timedeltas):
     return 'seconds'
 
 
+def infer_calendar_name(dates):
+    """Given an array of datetimes, infer the CF calendar name"""
+    if np.asarray(dates).dtype == 'datetime64[ns]':
+        return 'proleptic_gregorian'
+    else:
+        try:
+            return np.asarray(dates)[0].calendar
+        except IndexError:
+            return np.asarray(dates).item().calendar
+
+
 def infer_datetime_units(dates):
     """Given an array of datetimes, returns a CF compatible time-unit string of
     the form "{time_unit} since {date[0]}", where `time_unit` is 'days',
     'hours', 'minutes' or 'seconds' (the first one that can evenly divide all
     unique time deltas in `dates`)
     """
-    dates = pd.to_datetime(np.asarray(dates).ravel(), box=False)
-    dates = dates[pd.notnull(dates)]
-    unique_timedeltas = np.unique(np.diff(dates))
+    if np.asarray(dates).dtype == 'datetime64[ns]':
+        dates = pd.to_datetime(np.asarray(dates).ravel(), box=False)
+        dates = dates[pd.notnull(dates)]
+        unique_timedeltas = np.unique(np.diff(dates))
+        reference_date = dates[0] if len(dates) > 0 else '1970-01-01'
+        reference_date = pd.Timestamp(reference_date)
+    else:
+        if not OPTIONS['enable_cftimeindex']:
+            raise ValueError('Serializing dates of type cftime.datetime '
```

Comment:
I don't think we need to raise an error here, since this wouldn't change any existing behavior  -- nobody is relying on errors being raised when they attempt to save arrays of netcdftime datetimes to netcdf files!

We really only need to check the option for creating a CFTimeIndex automatically when loading a file from disk.